smart-connector-api/src/main/java/eu/knowledge/engine/smartconnector/api/AskResult.java

```diff
@@ -99,8 +103,12 @@ public ReasonerPlan getReasonerPlan() {
 		return this.reasonerPlan;
 	}
 
+	public Set<KnowledgeGap> getKnowledgeGaps() {
+		return this.knowledgeGaps;
+	}
```

Comment:
@JackJackie Something to consider is to not only make knowledge gaps available on `AskResult`, but also on `AskPlan`. In some scenario's you want to determine knowledge gaps without actually executing the `AskKnowledgeInteraction`. The `admin-ui` could, for example, use this to show which knowledge interactions have knowledge gaps, but we do not want to have to execute the `AskKnowledgeInteraction` before determining the knowledge gaps.

Since the REST API does not distinguish between `planning` and `executing`, I think only the Java API should support this. Do you think we should include this to this Pull Request (PR), or do we just want to finally close it?

Reply:
All right @bnouwt : let's discuss when you have the opportunity!

Reply:
Decided to add this as a separate issue: https://github.com/TNO/knowledge-engine/issues/561